### PR TITLE
IIS beta to ga

### LINKED ISSUE #3523: Make IIS GA
# Integration release checklist

### All changes

- [x] Change follows the [contributing guidelines](https://github.com/elastic/integrations/blob/main/CONTRIBUTING.md)
- [x] Supported versions of the monitoring target are documented
- [x] Supported operating systems are documented (if applicable)
- [x] Documentation exists
- [x] Fields follow [ECS](https://github.com/elastic/ecs) and [naming conventions](https://www.elastic.co/guide/en/beats/devguide/master/event-conventions.html)
- [x] At least a manual test with ES / Kibana / Agent has been performed.
- [x] Required Kibana version set to:

Tasks
- [x] Updated ECS version to 8.2.0 in build.yml
- [x] Remove duplicate entries from the fields.yml
- [x] Add sub fields for user_agent and geoip group fields in fields.yml
- [x] Change the visualization references to by value in Kibana dashboards
- [x] Add unit and metric_type for fields where ever necessary

Attaching the documentation [here](https://docs.google.com/document/d/1wogmk5M1W8Dzsyoui9MhCRrzBNitSjYQ0yCrciJxm3U/edit#)

### Dashboards changes

- [x] Dashboards exists

### Log dataset changes

- [x] [Pipeline tests](https://github.com/elastic/elastic-package/blob/master/docs/howto/pipeline_testing.md) exist (if applicable)
- [x] Generated output for at least 1 log file exists

### Metric dataset changes

_This entry is currently only recommended. It will be mandatory once we provide better support for it._

- [x] Sample event (`sample_event.json`) exists

### PR DESCRIPTION
**Pull request by muthu-mps**
## What does this PR do?
Moving the IIS integration from beta to GA. 

## Checklist

- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## How to test this PR locally
+ Install IIS server in the windows platform
+ Add IIS agent integration
+ Check whether the collect logs and metrics option is enabled
+ Verify the data is ingested into Elasticsearch
+ Check dashboards loads with the data and generates metrics

Since IIS is windows platform specific integration. Manual testing is performed and system testing is not added.

## Related issues

- Closes #3523

### CODE REVIEW

File: packages/iis/data_stream/access/fields/ecs.yml [hunk 1]
```diff
@@ -72,22 +72,14 @@
   name: user.name
 - external: ecs
   name: user_agent.device.name
-- external: ecs
-  name: user_agent.device.name
```
Comment (ishleenk17) [inline packages/iis/data_stream/access/fields/ecs.yml:76]: Why aren't we taking the ecs values for these pre indexed fields ?

File: packages/iis/data_stream/website/fields/fields.yml [hunk 2]
```diff
@@ -50,29 +58,38 @@
...
         - name: total_bytes_received
           type: float
+          unit: byte
+          metric_type: gauge
           description: |
             The total number of bytes received.
         - name: total_bytes_sent
           type: float
+          unit: byte
+          metric_type: gauge
```
Comment (ishleenk17) [inline packages/iis/data_stream/website/fields/fields.yml:68]: This is a value that will always be increasing. Won't be changing up and down. 
IMO, this should be counter.

File: packages/iis/data_stream/website/fields/fields.yml [hunk 3]
```diff
@@ -50,29 +58,38 @@
             The service uptime.
         - name: total_bytes_received
           type: float
+          unit: byte
+          metric_type: gauge
```
Comment (ishleenk17) [inline packages/iis/data_stream/website/fields/fields.yml:62]: This is a value that will always be increasing. Won't be changing up and down.
IMO, this should be counter.

File: packages/iis/data_stream/webserver/fields/fields.yml [hunk 4]
```diff
@@ -198,14 +220,19 @@
             Service uptime.
         - name: total_anonymous_users
           type: float
+          metric_type: counter
           description: |
             Total number of anonymous users.
         - name: total_bytes_received
           type: float
+          unit: byte
+          metric_type: gauge
```
Comment (ishleenk17) [inline packages/iis/data_stream/webserver/fields/fields.yml:229]: This is a value that will always be increasing. Won't be changing up and down.
IMO, this should be counter.

File: packages/iis/data_stream/webserver/fields/fields.yml [hunk 5]
```diff
@@ -198,14 +220,19 @@
...
           description: |
             Total size of bytes received.
         - name: total_bytes_sent
           type: float
+          unit: byte
+          metric_type: gauge
```
Comment (ishleenk17) [inline packages/iis/data_stream/webserver/fields/fields.yml:235]: This is a value that will always be increasing. Won't be changing up and down.
IMO, this should be counter.

File: packages/iis/data_stream/website/fields/fields.yml (hunk 2 above)
Comment (muthu-mps) [reply packages/iis/data_stream/website/fields/fields.yml]: done.

File: packages/iis/data_stream/access/fields/ecs.yml (hunk 1 above)
Comment (muthu-mps) [reply packages/iis/data_stream/access/fields/ecs.yml:76]: done.

File: packages/iis/data_stream/webserver/fields/fields.yml (hunk 5 above)
Comment (muthu-mps) [reply packages/iis/data_stream/webserver/fields/fields.yml]: done.

File: packages/iis/data_stream/webserver/fields/fields.yml (hunk 4 above)
Comment (muthu-mps) [reply packages/iis/data_stream/webserver/fields/fields.yml]: done.

File: packages/iis/data_stream/access/_dev/test/pipeline/test-iis-8.log [hunk 6]
```diff
@@ -0,0 +1,6 @@
+2019-08-14 00:09:32 W3SVC1 SE119654 ::1 GET /help/default.aspx area= 80 - ::1 HTTP/1.1 Mozilla/5.0+(compatible;+MSIE+9.0;+Windows+NT+6.1;+Trident/5.0) - - localhost 403 4 5 5279 160 15
+2019-08-14 00:09:32 W3SVC1 SE119654 ::1 GET /help/default.aspx area= 80 - ::1 HTTP/1.1 Mozilla/5.0+(compatible;+MSIE+9.0;+Windows+NT+6.1;+Trident/5.0) - - localhost 403 4 5 5279 136 0
+2019-08-14 00:09:32 W3SVC1 SE119654 ::1 GET /help/default.aspx area= 80 - ::1 HTTP/1.1 Mozilla/5.0+(compatible;+MSIE+9.0;+Windows+NT+6.1;+Trident/5.0) - - localhost 403 4 5 5279 136 0
+2019-08-14 00:12:02 W3SVC1 SE119654 10.60.79.142 GET /default.aspx - 80 - 10.60.74.238 HTTP/1.0 - - - - 403 4 5 1372 18 0
+2019-08-14 00:13:10 W3SVC1 SE119654 10.60.79.142 GET /default.aspx - 80 - 10.60.74.238 HTTP/1.1 Mozilla/4.0+(compatible;+MSIE+8.0;+Windows+NT+5.1;+Trident/4.0) - - se119654.saifg.rbc.com 403 4 5 1353 313 15
```
Comment (ishleenk17) [inline packages/iis/data_stream/access/_dev/test/pipeline/test-iis-8.log:5]: I don't see the ingest pipeline patterns for this format. 
Do they already exist ?

File: packages/iis/data_stream/access/_dev/test/pipeline/test-iis-8.log-expected.json [hunk 7]
```diff
@@ -0,0 +1,466 @@
...
+                "version": "8.2.1"
+            },
+            "event": {
+                "category": [
+                    "web",
+                    "network"
+                ],
+                "kind": "event",
+                "original": "2019-08-14 00:09:32 W3SVC1 SE119654 ::1 GET /help/default.aspx area= 80 - ::1 HTTP/1.1 Mozilla/5.0+(compatible;+MSIE+9.0;+Windows+NT+6.1;+Trident/5.0) - - localhost 403 4 5 5279 160 15",
+                "outcome": "failure",
```
Comment (lalit-satapathy) [inline packages/iis/data_stream/access/_dev/test/pipeline/test-iis-8.log-expected.json:21]: what causes this event.outcome as "failure"?

File: packages/iis/docs/README.md [hunk 8]
```diff
@@ -109,97 +109,97 @@ The fields reported are:
```
Comment (lalit-satapathy) [inline packages/iis/docs/README.md:109]: May be we can add a summary line explicitly on the log format types supported by IIS (W3C) in the corresponding section.

**Review by lalit-satapathy — COMMENTED** [review]
Are there any changes to the dashboard image files?

File: packages/iis/data_stream/access/_dev/test/pipeline/test-iis-8.log-expected.json (hunk 7 above)
Comment (muthu-mps) [reply packages/iis/data_stream/access/_dev/test/pipeline/test-iis-8.log-expected.json:21]: The sample log which we are processing is with `http_status_code: 403`.  Based on the status code the outcome is marked as failure.

**Comment by muthu-mps** [comment]
> Are there any changes to the dashboard image files?

The only change is dashboard by reference changed to by-value.

File: packages/iis/_dev/build/docs/README.md [hunk 9]
```diff
@@ -48,7 +48,7 @@ The fields reported are:
 The IIS module has been tested with logs from version 7.5 and version 10.
```
Comment (ishleenk17) [inline packages/iis/_dev/build/docs/README.md:48]: I think we support IIS logs of version 8 as well. Mention that in the readme

**Comment by ishleenk17** [comment]
I think it would be worth mentioning somewhere in a document (maybe under _dev) as to why system tests are not present.
We can link it to the issue where system tests not supported for windows issue is open.

**Review by ishleenk17 — APPROVED** [review]
Looks good !

**Pull request merged**